core/src/main/java/org/apache/carbondata/core/scan/executor/impl/AbstractQueryExecutor.java

```diff
@@ -230,6 +230,11 @@ protected void initQuery(QueryModel queryModel) throws IOException {
           updateColumns(queryModel, fileFooter.getColumnInTable(), blockInfo.getFilePath());
           filePathToSegmentPropertiesMap.put(blockInfo.getFilePath(), segmentProperties);
         }
+        //In case of Legacy store, set setDirectVectorFill to false,
+        // as filling vector directly is not supported for V2 store
```

Comment:
not only v2... 'not supported for legacy store prior to v3'. Besides I do like you to add a warn log here and remove the comments.